File: kotlin-swift-interopedia-samples-without-skie/shared/src/commonMain/kotlin/com/jetbrains/swiftinteropplayground/classes/OpenClass.kt
package com.jetbrains.swiftinteropplayground.classes

open class OpenClassWithConstructorParams(
    val param1: String,
    val param2: Boolean
) {
    protected val someField: String get() = "14"

    fun finalFunctionInClass(): String = "Final method in open class"

    open fun functionCanBeOverridden(): String = "Open method in open class"
}

File: kotlin-swift-interopedia-samples-without-skie/iosApp/iosApp/Classes/OpenClass.swift
import Foundation
import shared

class OpenClassExample: OpenClassWithConstructorParams {
//    // Overriding leads to an exception at runtime on instantiation 
//    override func finalFunctionInClass() -> String {
//        "Final function can't be overridden"
//    }

    override func functionCanBeOverridden() -> String {
        "Open function overridden from Swift"
    }
}

func openClassExample() {
    let openClass = OpenClassExample(param1: "123", param2: true)

    print("Accessing field: \(openClass.someField)")
    print("Calling \(openClass.finalFunctionInClass())")
    print("Calling \(openClass.functionCanBeOverridden())")
}